### PR TITLE
feat(Templates): Restructure to Reflect Custom Templates Restructuring

### PR DESCRIPTION
**Pull request by Elaina-Lee**
## Type of Change

* [ ] Bug fix
* [X] Feature
* [ ] Other

- No feature updates (No change in UI)
- Restructuring templates to align with manifest separation of TemplateManifest and WorkflowManifest
- Introduction: id for both manifests, description => summary, detailsDescription => description
- Mock data updates for LocalTemplates

## Impact of Change

Restructuring: Separating Template.Manifest into Template.TemplateManifest and Template.WorkflowManifest
- Disregarding of whether a wrkflow or accelerator template, it will follow the same way of containing manifests

* [ ] **This is a breaking change.**

## Test Plan

All previous tests using Template.Manifest has been updated and have been verified to do the same with new way of redux stores updates containing separate manifests

### CODE REVIEW

File: libs/logic-apps-shared/src/utils/src/lib/models/template.ts [hunk 1]
```diff
@@ -1,29 +1,48 @@
 export type SkuType = 'standard' | 'consumption';
 export type WorkflowKindType = 'stateful' | 'stateless';
-export type ConnectorRuntimeType = 'inapp' | 'shared';
+export type ConnectorRuntimeType = 'inapp' | 'shared' | 'custom';
```
Comment (preetriti1) [inline libs/logic-apps-shared/src/utils/src/lib/models/template.ts:3]: Why do you have custom here?

File: libs/logic-apps-shared/src/designer-client-services/lib/base/template.ts [hunk 2]
```diff
@@ -34,35 +34,39 @@ export class BaseTemplateService implements ITemplateService {
...
+        uri: `${endpoint}/templates/${templateId}/manifest.json`,
         headers: { 'Access-Control-Allow-Origin': '*' },
       });
     }
 
-    const paths = resourcePath.split('/');
-
-    return paths.length === 2
-      ? (await import(`./../templates/${paths[0]}/${paths[1]}/manifest.json`)).default
-      : (await import(`./../templates/${resourcePath}/manifest.json`)).default;
```
Comment (preetriti1) [inline libs/logic-apps-shared/src/designer-client-services/lib/base/template.ts:50]: Did we create two different methods to avoid this path split check? We can just use resourceManifest<T> based on the type being returned.
We can create multiple methods inside designer helper functions/actions but in services it is better to allow less implementation from host. I dont see why we need two different functions to download from same path and same manifest.json

File: libs/designer/src/lib/ui/templates/templateoverview.tsx [hunk 3]
```diff
@@ -157,25 +157,23 @@ export const TemplateOverview = ({
 interface WorkflowItem {
   id: string;
   name: string;
-  trigger: string;
```
Comment (preetriti1) [inline libs/designer/src/lib/ui/templates/templateoverview.tsx:160]: Why this is removed after a template selection? This was always being read from workflow.json when template selection was happening no?

File: libs/logic-apps-shared/src/utils/src/lib/models/template.ts (hunk 1 above)
Comment (Elaina-Lee) [reply libs/logic-apps-shared/src/utils/src/lib/models/template.ts:3]: Kind allowed on templates repo is 'inapp', 'shared', and 'custom'. Do we not planning on supporting this - should this be removed from all?

File: libs/logic-apps-shared/src/designer-client-services/lib/base/template.ts (hunk 2 above)
Comment (Elaina-Lee) [reply libs/logic-apps-shared/src/designer-client-services/lib/base/template.ts:50]: Original idea was - those two return a different type (TemplateManifest, WorkflowManifest). I can do the type of OR, but that means we would have to we have to infer the type `as TemplateManifest` everytime. Also, for workflowManifest, we can clarify templateId and workflowId.

In that case, I can change it back.

File: libs/designer/src/lib/ui/templates/templateoverview.tsx (hunk 3 above)
Comment (Elaina-Lee) [reply libs/designer/src/lib/ui/templates/templateoverview.tsx:160]: No this was not being read from workflow.json. If you look at the below code, it was doing `const trigger = getPropertyValue(details, 'trigger');` which is exactly the point I brought up in the chat. WorkflowManifest doesn't have "Trigger" in our contract - how should we approach this?

File: libs/logic-apps-shared/src/designer-client-services/lib/template.ts [hunk 4]
```diff
@@ -6,9 +6,9 @@ export interface ITemplateService {
   openBladeAfterCreate: (workflowName: string | undefined) => void;
   onAddBlankWorkflow: () => Promise<void>;
   getAllTemplateNames: () => Promise<string[]>;
-  getResourceManifest: (resourcePath: string) => Promise<Template.Manifest>;
-  getWorkflowDefinition: (resourcePath: string) => Promise<LogicAppsV2.WorkflowDefinition>;
-  getContentPathUrl: (templateName: string, resourcePath: string) => string;
+  getResourceManifest: (templateId: string) => Promise<Template.TemplateManifest | Template.WorkflowManifest>;
```
Comment (preetriti1) [inline libs/logic-apps-shared/src/designer-client-services/lib/template.ts:9]: lets call it resourcePath

File: libs/designer/src/lib/ui/templates/templateoverview.tsx (hunk 3 above)
Comment (preetriti1) [reply libs/designer/src/lib/ui/templates/templateoverview.tsx:160]: I was expecting us to save this data in the custom template [ this will be inferred data from workflow.json ], so github templates would still have this data in the details of templates... and when we publish the template in production we can always infer and put it there.

File: libs/designer/src/lib/ui/templates/__tests__/templateCard.spec.tsx [hunk 5]
```diff
@@ -11,63 +11,68 @@ import React from 'react';
...
+        By: '',
+        Type: '',
+        Category: '',
+      },
       artifacts: [
         {
-          type: 'workflow',
-          file: 'workflow.json',
+          type: 'description',
+          file: 'description.md',
```
Comment (preetriti1) [inline libs/designer/src/lib/ui/templates/__tests__/templateCard.spec.tsx:35]: We never have this... this portion would include the maps/schemas right? are you trying to test anything in that aspect? if not then just leave it empty

File: libs/designer/src/lib/ui/templates/__tests__/displayConnections.spec.tsx [hunk 6]
```diff
@@ -13,31 +13,66 @@ import React from 'react';
...
+      },
+      details: {
+        By: '',
+        Type: '',
+        Category: '',
+      },
+      artifacts: [
+        {
+          type: 'description',
+          file: 'description.md',
```
Comment (preetriti1) [inline libs/designer/src/lib/ui/templates/__tests__/displayConnections.spec.tsx:40]: Why are we not keeping this empty in test files... is something breaking? also artifacts was not a required property right so if not present we should not fail in our code

File: libs/designer/src/lib/ui/panel/templatePanel/quickViewPanel/tabs/summaryTab.tsx [hunk 7]
```diff
@@ -33,7 +34,7 @@ export const SummaryPanel = ({ workflowId }: { workflowId: string }) => {
     }),
   };
 
-  return isNullOrUndefined(manifest) ? null : (
+  return isNullOrUndefined(workflowManifest) || isNullOrUndefined(templateManifest) ? null : (
```
Comment (preetriti1) [inline libs/designer/src/lib/ui/panel/templatePanel/quickViewPanel/tabs/summaryTab.tsx:37]: This is just a fail safe check right?

File: libs/designer/src/lib/ui/panel/templatePanel/quickViewPanel/tabs/summaryTab.tsx [hunk 8]
```diff
@@ -88,22 +89,22 @@ export const SummaryPanel = ({ workflowId }: { workflowId: string }) => {
...
-              <Text className="msla-template-overview-section-detailkey">{detailsTags[key]}:</Text>
-              <Text>{manifest.details[key]}</Text>
+              <Text className="msla-template-overview-section-detailkey">{detailsTags[key as Template.DetailsType]}:</Text>
+              <Text>{templateManifest.details[key as Template.DetailsType]}</Text>
             </div>
           );
         })}
       </div>
-      {manifest.tags?.length ? (
+      {templateManifest.tags?.length ? (
```
Comment (preetriti1) [inline libs/designer/src/lib/ui/panel/templatePanel/quickViewPanel/tabs/summaryTab.tsx:107]: SO both for accelerators and single workflow templates we will always be showing the template tags here not the workflow specific tags... wont we have tags in both the locations?

File: libs/designer/src/lib/ui/panel/templatePanel/quickViewPanel/quickViewPanel.tsx [hunk 9]
```diff
@@ -72,12 +75,12 @@ export const QuickViewPanel = ({
     () => (
       <QuickViewPanelHeader
         title={manifest.title}
-        description={manifest.description}
+        summary={manifest.summary}
         sourceCodeUrl={manifest.sourceCodeUrl}
-        details={manifest?.details ?? {}}
+        details={templateManifest?.details ?? {}}
       />
```
Comment (preetriti1) [inline libs/designer/src/lib/ui/panel/templatePanel/quickViewPanel/quickViewPanel.tsx:81]: Wont we need to read the trigger from workflow manifest or is it not needed in header?

File: libs/designer/src/lib/core/actions/bjsworkflow/templates.ts [hunk 10]
```diff
@@ -37,10 +37,13 @@ import {
...
-  manifest: Template.Manifest;
+  manifest: Template.WorkflowManifest;
   workflowName: string | undefined;
   kind: string | undefined;
-  images?: Record<string, string>;
+  images?: {
+    light?: string;
+    dark?: string;
+  };
   connectionKeys: string[];
```
Comment (preetriti1) [inline libs/designer/src/lib/core/actions/bjsworkflow/templates.ts:47]: Didn't we have the parameterKeys also here? 
If not how were we finding the associated workflows for each parameter, was that directly looking into manifest.parmeters?

File: libs/designer/src/lib/core/actions/bjsworkflow/templates.ts [hunk 11]
```diff
@@ -244,16 +248,12 @@ const loadTemplateFromResourcePath = async (
   };
 
   if (isMultiWorkflow && workflows) {
```
Comment (preetriti1) [inline libs/designer/src/lib/core/actions/bjsworkflow/templates.ts:250]: Since workflows is a required property now, lets remove the check

File: libs/designer/src/lib/core/actions/bjsworkflow/templates.ts [hunk 12]
```diff
@@ -281,11 +281,19 @@ const loadTemplateFromResourcePath = async (
     }
   } else {
```
Comment (preetriti1) [inline libs/designer/src/lib/core/actions/bjsworkflow/templates.ts:282]: Can we remove this else section now? workflows is required and we dont have to explicitly check for multiworkflows since now the workflowId is also marked with default.

File: libs/designer/src/lib/core/templates/utils/helper.ts [hunk 13]
```diff
@@ -53,23 +53,30 @@ export const getUniqueConnectors = (
...
 ): Template.Connection[] => {
-  const allConnections = Object.values(connections);
+  const allConnections = Object.values(connections).map((connection) => ({ ...connection, id: connection.connectorId }));
   return getUniqueConnectorsFromConnections(allConnections, subscriptionId, location);
 };
 
+interface UniqueConnectorInfo {
+  connectorId: string;
+  operationId: string | undefined;
+}
```
Comment (preetriti1) [inline libs/designer/src/lib/core/templates/utils/helper.ts:63]: Dont we already have an interface called OperationInfo from for this..
and why are we including operationId in uniqueconnectorInfo

File: libs/designer/src/lib/core/templates/utils/helper.ts [hunk 14]
```diff
@@ -53,23 +53,30 @@ export const getUniqueConnectors = (
   subscriptionId: string,
   location: string
 ): Template.Connection[] => {
-  const allConnections = Object.values(connections);
+  const allConnections = Object.values(connections).map((connection) => ({ ...connection, id: connection.connectorId }));
```
Comment (preetriti1) [inline libs/designer/src/lib/core/templates/utils/helper.ts:56]: I got really confused here - please change the variable name to featuredConnectors

**Review by preetriti1 — CHANGES_REQUESTED** [review]

File: libs/designer/src/lib/ui/panel/templatePanel/quickViewPanel/tabs/summaryTab.tsx (hunk 7 above)
Comment (Elaina-Lee) [reply libs/designer/src/lib/ui/panel/templatePanel/quickViewPanel/tabs/summaryTab.tsx:37]: Yes, otherwise we'd need to do null check for reading fields for the rendering for each section

File: libs/designer/src/lib/core/actions/bjsworkflow/templates.ts (hunk 12 above)
Comment (Elaina-Lee) [reply libs/designer/src/lib/core/actions/bjsworkflow/templates.ts:282]: Are we going to check `isMultiWorkflow` for every loop for `workflowData.workflow.workflowName = workflows[workflowId].name;` then?

I thought it was a bit unpractical to have the reduce function and spread for single workflow case... but we could do that as well

File: libs/designer/src/lib/core/templates/utils/helper.ts (hunk 13 above)
Comment (Elaina-Lee) [reply libs/designer/src/lib/core/templates/utils/helper.ts:63]: Because OperationInfo requires that operationId is a string not undefined

File: libs/designer/src/lib/core/actions/bjsworkflow/templates.ts (hunk 10 above)
Comment (Elaina-Lee) [reply libs/designer/src/lib/core/actions/bjsworkflow/templates.ts:47]: Associated workflows is present under parameter definition, so it can be read directly. We don't look into manifest.parameters.

File: libs/designer/src/lib/core/templates/utils/helper.ts (hunk 13 above)
Comment (Elaina-Lee) [reply libs/designer/src/lib/core/templates/utils/helper.ts:63]: We weren't using OperationInfo since there operationId is a required property. Removed this since we won't use operationId as per our yesterday discussion about connector icons

File: libs/designer/src/lib/core/templates/utils/helper.ts (hunk 14 above)
Comment (Elaina-Lee) [reply libs/designer/src/lib/core/templates/utils/helper.ts]: Changed to allConnectors; We use this for `connections.tsx` where we combine all connections on the summary tab (or overview page in case of accelerator) - which is not featured connectors but all connectors

File: libs/designer/src/lib/ui/templates/templateoverview.tsx (hunk 3 above)
Comment (Elaina-Lee) [reply libs/designer/src/lib/ui/templates/templateoverview.tsx:160]: Discussed offline

File: libs/designer/src/lib/ui/panel/templatePanel/quickViewPanel/quickViewPanel.tsx (hunk 9 above)
Comment (Elaina-Lee) [reply libs/designer/src/lib/ui/panel/templatePanel/quickViewPanel/quickViewPanel.tsx:81]: Discussed offline; trigger not present in workflowManifest

File: libs/designer/src/lib/core/templates/utils/helper.ts [hunk 15]
```diff
@@ -52,24 +52,26 @@ export const getUniqueConnectors = (
...
   location: string
-): Template.Connection[] => {
-  const allConnections = Object.values(connections);
-  return getUniqueConnectorsFromConnections(allConnections, subscriptionId, location);
+): Template.FeaturedConnector[] => {
+  const allConnectors = Object.values(connections).map((connector) => ({ ...connector, id: connector.connectorId }));
+  return getUniqueConnectorsFromConnections(allConnectors, subscriptionId, location);
 };
 
 export const getUniqueConnectorsFromConnections = (
```
Comment (preetriti1) [inline libs/designer/src/lib/core/templates/utils/helper.ts:60]: Can you double check if this is not getting called to list the connections in teh connections grid and also showing the connections summary in templateoverview and workflow summary tab?

File: libs/logic-apps-shared/src/utils/src/lib/models/template.ts [hunk 16]
```diff
@@ -1,29 +1,48 @@
 export type SkuType = 'standard' | 'consumption';
 export type WorkflowKindType = 'stateful' | 'stateless';
-export type ConnectorRuntimeType = 'inapp' | 'shared';
+export type ConnectorRuntimeType = 'inapp' | 'shared' | 'builtin';
+export type FeaturedConnectorType = ConnectorRuntimeType | 'builtin';
```
Comment (preetriti1) [inline libs/logic-apps-shared/src/utils/src/lib/models/template.ts:4]: These both are always same right? Then why do we need a separate type?

Comment (Elaina-Lee) [reply libs/logic-apps-shared/src/utils/src/lib/models/template.ts]: Removed 'builtin' from ConnectorRuntimeType

**Review by preetriti1 — APPROVED** [review]
